.github/workflows/docker_push.yml

```diff
@@ -0,0 +1,64 @@
+# GitHub actions workflow which builds and publishes the docker images.
+
+name: Build and push docker images
+
+on:
+  push:
+    tags: ["v*"]
+    # uncomment this to enable all pushes to main
+    #branches: [ master, main ]
```

Comment:
I think `latest` on our docker images always just points to the latest release. Not sure if we'd ever push on the `main` branch.

Though you mention re-enabling building docker images on main below - did we used to do that?

Reply:
I ripped off https://github.com/matrix-org/synapse/blob/develop/.github/workflows/docker.yml which seems like if you push to master on Synapse, you will get a new build.

I'm not really sure why this is, so I took it out for Sygnal.

Reply:
Given we don't have anything other than releases on [our dockerhub](https://hub.docker.com/r/matrixdotorg/synapse/tags?page=1&ordering=last_updated), I believe that that workflow both tests docker builds as well as publishes them (if the branch has an associated release tag).

This isn't necessary in Sygnal, as we already have a workflow for testing docker builds.